resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/submit/submitsteps/MountSmallLocalFilesStep.scala

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.deploy.kubernetes.submit.submitsteps
+
+import java.io.File
+
+import com.google.common.io.{BaseEncoding, Files}
+import io.fabric8.kubernetes.api.model.SecretBuilder
+import scala.collection.JavaConverters._
+
+import org.apache.spark.deploy.kubernetes.config._
+import org.apache.spark.deploy.kubernetes.submit.{KubernetesFileUtils, MountSmallFilesBootstrap}
+import org.apache.spark.util.Utils
+
+private[spark] class MountSmallLocalFilesStep(
+    sparkFiles: Seq[String],
+    smallFilesSecretName: String,
+    smallFilesSecretMountPath: String,
+    mountSmallFilesBootstrap: MountSmallFilesBootstrap) extends DriverConfigurationStep {
+
+  private val MAX_SECRET_BUNDLE_SIZE_BYTES = 10000
+  private val MAX_SECRET_BUNDLE_SIZE_BYTES_STRING =
+      Utils.bytesToString(MAX_SECRET_BUNDLE_SIZE_BYTES)
+
+  override def configureDriver(driverSpec: KubernetesDriverSpec): KubernetesDriverSpec = {
+    val localFiles = KubernetesFileUtils.getOnlySubmitterLocalFiles(sparkFiles).map(new File(_))
+    val totalSizeBytes = localFiles.map(_.length()).sum
```

Comment:
Yeah, the reasoning here is that it's clearer to message "Your files are X bytes, max is Y bytes" as opposed to "Your base64-encoded files are X bytes, max base-64 encoded is Y bytes". Users can directly check sizes of files far more easily than base64-encoded files.